### PR TITLE
Feature: admin table functions

### PR DESCRIPTION
**Pull request by edbrett**
Rebuild of the admin tables to unify functions for front and backend with the following:
- Add pagination to tables
- Sort functions for selected columns
- Styles to remove dependancy on bootstrap
- Live search functions for all tables

### CODE REVIEW

File: app/controllers/admin/campaigns_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class Admin::CampaignsController < AdminController
   # GET /campaigns
   # GET /campaigns.json
   def index
-    @campaigns = Campaign.search(params[:search]).order(sort_column + " " + sort_direction).paginate(:page => params[:page], :per_page => 10)
+    @campaigns = Campaign.search(params[:search]).order(sort_column + " " + sort_direction).paginate(:page => params[:page], :per_page => 9)
```
Comment (simaob) [inline app/controllers/admin/campaigns_controller.rb]: Can you standardise the hash syntax all across? let's use the `{ new: 'syntax' }`

File: app/controllers/admin/countries_controller.rb [hunk 2]
```diff
@@ -71,4 +75,13 @@ def set_country
     def country_params
       params.require(:country).permit(:name, :iso, :shp_url, :geojson_url)
     end
+
+  private
+    def sort_column
+      Country.column_names.include?(params[:sort]) ? params[:sort] : "name"
+    end
+
+    def sort_direction
```
Comment (simaob) [inline app/controllers/admin/countries_controller.rb]: as discussed you could move this method to the AdminController and would be available here and in the other controllers that inherit from it.

File: app/controllers/admin/countries_controller.rb [hunk 3]
```diff
@@ -71,4 +75,13 @@ def set_country
     def country_params
       params.require(:country).permit(:name, :iso, :shp_url, :geojson_url)
     end
+
+  private
+    def sort_column
```
Comment (simaob) [inline app/controllers/admin/countries_controller.rb]: For this method you could do the same as sort_direction, and you could get the model name by doing: `controller_name.classify.constantize`

File: app/views/admin/campaigns/_campaigns.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  <tbody>
+    <% for campaign in @campaigns %>
+      <tr>
+        <td class="-bold"><%= campaign.name %></td>
+        <td><%= campaign.description %></td>
+        <td><%= campaign.htag %></td>
+        <td><%= campaign.start_date %></td>
+        <td><%= campaign.status_humanize %></td>
+        <td class="actions">
+          <%= link_to '', campaign_path(campaign), :class => "svg-bttn -show" %>
```
Comment (simaob) [inline app/views/admin/campaigns/_campaigns.html.erb]: As we are linking to the public interface here, and there's no easy way to come back here, I'd open this and other show links in a new tab. Unless your goal was to link to the show page of the admin, then you'll need to update the path you are using.

Comment (edbrett) [reply app/views/admin/campaigns/_campaigns.html.erb]: Agreed. Juan Carlos preferred linking this way as we already have a view.

**Pull request merged**